src/query/service/src/interpreters/interpreter_insert_v2.rs

```diff
@@ -154,6 +156,51 @@ impl InsertInterpreterV2 {
         }
     }
 
+    async fn prepared_values(&self, values_str: &str) -> Result<(DataSchemaRef, Vec<DataScalar>)> {
+        let settings = self.ctx.get_settings();
+        let sql_dialect = settings.get_sql_dialect()?;
+        let tokens = tokenize_sql(values_str)?;
+        let backtrace = Backtrace::new();
+        let expr_or_placeholders =
+            parser_values_with_placeholder(&tokens, sql_dialect, &backtrace)?;
+        let source_schema = self.plan.schema();
+
+        if source_schema.num_fields() != expr_or_placeholders.len() {
+            return Err(ErrorCode::SemanticError(format!(
+                "need {} fields in values, got only {}",
+                source_schema.num_fields(),
+                expr_or_placeholders.len()
+            )));
+        }
+
+        let mut attachment_fields = vec![];
+        let mut const_fields = vec![];
+        let mut exprs = vec![];
+        for (i, eo) in expr_or_placeholders.into_iter().enumerate() {
+            match eo {
+                Some(e) => {
+                    exprs.push(e);
+                    const_fields.push(source_schema.fields()[i].clone());
+                }
+                None => attachment_fields.push(source_schema.fields()[i].clone()),
+            }
+        }
+        let name_resolution_ctx = NameResolutionContext::try_from(settings.as_ref())?;
+        let bind_context = BindContext::new();
+        let metadata = Arc::new(RwLock::new(Metadata::default()));
+        let const_schema = Arc::new(DataSchema::new(const_fields));
+        let const_values = exprs_to_scalar(
+            exprs,
+            &const_schema,
+            self.ctx.clone(),
+            &name_resolution_ctx,
+            &bind_context,
+            metadata,
+        )
+        .await?;
+        Ok((Arc::new(DataSchema::new(attachment_fields)), const_values))
+    }
+
     async fn build_insert_from_stage_pipeline(
```

Comment:
I think the place holder should work in other `InsertInputSource`.

```
StreamingWithFormat
StreamingWithFileFormat
```

Reply:
will be a little strange, like `insert into t1 values (?, 3, ?) ,(1, 2) (3, 4)`?

Reply:
`insert into t1 values (?, 3, ?)  format csv 1, 2\n3, 4`

Reply:
> will be a little strange, like `insert into t1 values (?, 3, ?) ,(1, 2) (3, 4)`?

This insert is invalid:
```
create table t1(a int, b int, c int);
insert into t1 values(1,2,3),(1,2); -- Inconsistent number of values in the VALUES clause expecting 3 but got 2
```

Reply:
yes,  it is example to demo that is `will be a little strange`,  need some desinge in gram to support it